diff -urN cgminer-3.10.0.org/compat/Makefile.am cgminer-3.10.0/compat/Makefile.am
--- cgminer-3.10.0.org/compat/Makefile.am	2013-11-30 22:24:45.000000000 +0100
+++ cgminer-3.10.0/compat/Makefile.am	2014-01-19 19:43:30.709577832 +0100
@@ -1,5 +1,5 @@
 
-SUBDIRS	= jansson-2.5
+SUBDIRS	=
 
 if WANT_USBUTILS
 if WANT_STATIC_LIBUSB
diff -urN cgminer-3.10.0.org/configure.ac cgminer-3.10.0/configure.ac
--- cgminer-3.10.0.org/configure.ac	2014-01-09 06:38:58.000000000 +0100
+++ cgminer-3.10.0/configure.ac	2014-01-19 19:43:30.709577832 +0100
@@ -337,9 +337,6 @@
 
 AM_CONDITIONAL([WANT_STATIC_LIBUSB], [test x$dlibusb = xno])
 
-AC_CONFIG_SUBDIRS([compat/jansson-2.5])
-JANSSON_LIBS="compat/jansson-2.5/src/.libs/libjansson.a"
-
 PKG_PROG_PKG_CONFIG()
 
 if test "x$have_cgminer_sdk" = "xtrue"; then
@@ -360,6 +357,8 @@
 	[libcurl=$enableval]
 	)
 
+PKG_CHECK_MODULES([JANSSON], [jansson >= 2.5])
+
 if test "x$libcurl" != xno; then
 	if test "x$have_win32" != xtrue; then
 		PKG_CHECK_MODULES([LIBCURL], [libcurl >= 7.25.0], [AC_DEFINE([CURL_HAS_KEEPALIVE], [1], [Defined if version of curl supports keepalive.])],
@@ -422,6 +421,7 @@
 AC_DEFINE_UNQUOTED([CGMINER_PREFIX], ["$prefix/bin"], [Path to cgminer install])
 
 AC_SUBST(JANSSON_LIBS)
+AC_SUBST(JANSSON_CFLAGS)
 AC_SUBST(PTHREAD_FLAGS)
 AC_SUBST(DLOPEN_FLAGS)
 AC_SUBST(PTHREAD_LIBS)
diff -urN cgminer-3.10.0.org/Makefile.am cgminer-3.10.0/Makefile.am
--- cgminer-3.10.0.org/Makefile.am	2014-01-09 06:20:18.000000000 +0100
+++ cgminer-3.10.0/Makefile.am	2014-01-19 19:44:00.330518487 +0100
@@ -1,8 +1,6 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-JANSSON_CPPFLAGS= -I$(top_builddir)/compat/jansson-2.5/src -I$(top_srcdir)/compat/jansson-2.5/src
-
 if WANT_USBUTILS
 USBUTILS_CPPFLAGS = -I$(top_builddir)/compat/libusb-1.0/libusb -I$(top_srcdir)/compat/libusb-1.0/libusb
 else
@@ -18,7 +16,7 @@
 
 SUBDIRS		= lib compat ccan
 
-cgminer_CPPFLAGS = $(PTHREAD_FLAGS) -fno-strict-aliasing $(JANSSON_CPPFLAGS) $(USBUTILS_CPPFLAGS)
+cgminer_CPPFLAGS = $(PTHREAD_FLAGS) -fno-strict-aliasing $(JANSSON_CFLAGS) $(USBUTILS_CPPFLAGS)
 
 bin_PROGRAMS	= cgminer
 
